lib/bolt/transport/orch.rb

```diff
@@ -210,6 +210,10 @@ def batch_task(targets, task, arguments, options = {}, &callback)
         end
       end
 
+      def batch_task_with(_targets, _task, _target_mapping, _options = {})
+        raise NotImplementedError, "pcp transport does not support batch_task_with()"
```

Comment:
This will result in an unintuitive error message if users try to call `run_task_with()` when using this transport.